chain/chain/src/store.rs

```diff
@@ -2343,10 +2343,18 @@ impl<'a> ChainStoreUpdate<'a> {
         if !epoch_manager.is_next_block_epoch_start(&block_hash)? {
             return Ok(());
         }
-        let epoch_id = epoch_manager.get_epoch_id(&block_hash)?;
-        let shard_layout = epoch_manager.get_shard_layout(&epoch_id)?;
-        let prev_epoch_id = epoch_manager.get_prev_epoch_id(&block_hash)?;
-        let prev_shard_layout = epoch_manager.get_shard_layout(&prev_epoch_id)?;
+
+        // We are using the block header to get the epoch_id and shard_layout here as BlockHeader
+        // isn't garbage collected.
+        let block_info = epoch_manager.get_block_info(&block_hash)?;
```

Comment:
It's a bit confusing that the comment says we're using the headers and immediately in the first line after you get the block info :) Maybe clarify for which blocks we expect to have the info and for which only the headers?

Reply:
Updated comments with more detailed explanation.